examples/stable_diffusion_v2/ldm/modules/attention.py

```diff
@@ -142,6 +162,34 @@ def rearange_in(x):
         k = rearange_in(k)
         v = rearange_in(v)
 
+        if self.enable_flash_attention and q.shape[1] % 16 == 0 and k.shape[1] % 16 == 0:
+            out = self.flash_attention(q, k, v)
+        else:
+            out = self.custom_attention(q, k, v, mask)
+
+        def rearange_out(x):
```

Comment:
is there a reason we are having a closure here, or not having this function outside

Reply:
> is there a reason we are having a closure here, or not having this function outside

Just for the uniformity of the code style, I can expand this ‘closure’ if necessary